crates/bitwarden-crypto/src/error.rs

```diff
@@ -0,0 +1,58 @@
+use std::fmt::Debug;
+
+use thiserror::Error;
+
+use crate::fingerprint::FingerprintError;
+
+#[derive(Debug, Error)]
+pub enum CryptoError {
+    #[error("The provided key is not the expected type")]
+    InvalidKey,
+    #[error("The cipher's MAC doesn't match the expected value")]
+    InvalidMac,
+    #[error("Error while decrypting EncString")]
+    KeyDecrypt,
+    #[error("The cipher key has an invalid length")]
+    InvalidKeyLen,
```

Comment:
Just noticed now that we have `InvalidKeyLen` and `InvalidLen` variants, and they both seem to be used with `hkdf.expand()`, maybe they should be combined?